linkcheck results in "can only concatenate str (not "bytes") to str"

**Describe the bug**
Runnin make linkcheck can result in an "can only concatenate str (not "bytes") to str" error.

**To Reproduce**
Steps to reproduce the behavior:
```
$ sphinx-quickstart -q -pfoo -afoo
$ rm index.rst
$ echo 'https://www.x.org/releases/X11R7.7/doc/man/man7/X.7.xhtml#heading5' > index.rst
$ make linkcheck
```
results in
```
<elided>

(line    1) broken    https://www.x.org/releases/X11R7.7/doc/man/man7/X.7.xhtml#heading5 - can only concatenate str (not "bytes") to str

build finished with problems.
make: *** [Makefile:20: linkcheck] Error 1
```

**Expected behavior**
Successful linkcheck: https://www.x.org/releases/X11R7.7/doc/man/man7/X.7.xhtml#heading5 is a valid URL, with no redirect whatsoever.

**Your project**
N/A

**Screenshots**
N/A

**Environment info**
- OS: Arch Linux
- Python version: 3.7.4
- Sphinx version: 2.2.1
- Sphinx extensions:  N/A
- Extra tools: N/A

**Additional context**
N/A

--- a/CHANGES
+++ b/CHANGES
@@ -48,6 +48,7 @@ Bugs fixed
   supported LaTeX engines: ¶, §, €, ∞, ±, →, ‣, –, superscript and subscript
   digits go through "as is" (as default OpenType font supports them)
 * #6704: linkcheck: Be defensive and handle newly defined HTTP error code
+* #6806: linkcheck: Failure on parsing content
 * #6655: image URLs containing ``data:`` causes gettext builder crashed
 * #6584: i18n: Error when compiling message catalogs on Hindi
 * #6718: i18n: KeyError is raised if section title and table title are same

--- a/sphinx/builders/linkcheck.py
+++ b/sphinx/builders/linkcheck.py
@@ -59,6 +59,9 @@ def check_anchor(response: requests.requests.Response, anchor: str) -> bool:
     # Read file in chunks. If we find a matching anchor, we break
     # the loop early in hopes not to have to download the whole thing.
     for chunk in response.iter_content(chunk_size=4096, decode_unicode=True):
+        if isinstance(chunk, bytes):    # requests failed to decode
+            chunk = chunk.decode()      # manually try to decode it
+
         parser.feed(chunk)
         if parser.found:
             break

